flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,124 @@
+
+        response = self._request('post', '/api/auth/authenticate', data={
+            'username': config['username'],
+            'password': config['password']
+        })
+
+        if response.status_code == 200:
+            self.connected = True
+            logger.debug('Successfully logged into Flood')
+        else:
+            self.connected = False
+            raise plugin.PluginError('Incorrect username or password')
+
+        return self
+    
+    def add_torrent_urls(self, **kwargs):
+        if not self.connected:
+            raise plugin.PluginError('Flood is not connected.')
+
+        # Add's a default 'start': True to the request
+        if not kwargs.get('start'):
+            kwargs['start'] = True
+        
+        response = self._request('post', '/api/torrents/add-urls', json=kwargs)
+
+        if response.status_code == 200:
+            logger.debug('Successfully added torrent(s).')
+        else:
+            # There's no sanity to the codes returned by Flood. 
+            # Both return "code": -32602 and both return status 500.
+            if response.text.find('the input is not a valid.') > 0:
+                raise plugin.PluginError('Not a valid torrent.')
+            elif response.text.find('Info hash already used by another torrent.') > 0:
+                logger.debug('Torrent has already been added to Flood.')
+            else:
+                raise plugin.PluginError(f'Failed to add torrent to Flood. Error {response.status_code}.')
+    
+    def add_torrent_files(self,  **kwargs):
+        return False
+
+class OutputFlood:
+    schema = {
+       'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'path': {'type': 'string'},
+            'tags': {'type': 'array'},
+        },
+        'additionalProperties': False,
+    }
+
+    def add_entry(self, config: dict, task: Task, entry: Entry):
+        kwargs = {}
+
+        url = entry.get('url', None)
+        path = entry.render(entry.get('path', config.get('path', None)))
+        tags = entry.get('tags', config.get('tags', None))
+
+        if url:
+            kwargs['urls'] = [ url ]
+        if path:
+            kwargs['destination'] = path
+        # Because of the way that Flood wants things
+        # We've gotta check to see if the person has done
+        # a list in their config or not.
+        if tags:
+            if isinstance(tags, list):
+                kwargs['tags'] = [ entry.render(tag) for tag in tags ]
+            else:
+                kwargs['tags'] = [ entry.render(tags) ]
+
+        if task.manager.options.test:
+            logger.info('Flood Test Mode')
+            logger.info('Would add torrent to Flood with:')
+            logger.info('\tPath: {}', path)
+            logger.info('\tTags: {}', tags)
+        else:
+            self.flood.add_torrent_urls(**kwargs)
```

Comment:
Using `**kwargs` in python is pretty annoying. I'd go as far as to claim that user written code should not use it at all. Only frameworks.

Please consider refactoring this so that the argument content is clearly defined and typed. Type hinting is pretty good way to document signatures too.

`**kwargs` is as ambiguous as it can be